app/controllers/transactions_controller.rb

```diff
@@ -42,6 +42,25 @@ def destroy
     @transaction.destroy
   end
 
+  def copy
```

Comment:
@dissident instead new action use same `create` action but use some query parameter like `copy_transaction_id` and if this parameter exists then prefill builded transaction with data from that transaction.